### PR TITLE
Update to Cluster Position Dependent Correction

### PR DESCRIPTION
**Pull request by ahodges21**
[comment]: <> (Please tell us something about this pull request)

This update adds CDB functionality to the residual correction of the position dependent correction.

## Types of changes
[comment]: <> ( What types of changes does your code introduce? Put an `x` in all the boxes that apply: )
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work for users)
- [ ] Requiring change in macros repository (Please provide links to the macros pull request in the last section)
- [x] I am a member of [GitHub organization of sPHENIX Collaboration](https://github.com/orgs/sPHENIX-Collaboration/people), EIC, or ECCE (contact Chris Pinkenburg to join)

## What kind of change does this PR introduce? (Bug fix, feature, ...)

[comment]: <> ( What does this PR do? Linking to talk in software meeting encouraged )

## TODOs (if applicable)

[comment]: <> ( In case this is a draft PR, e.g. for running checks using Jenkins, please make the pull request as a draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/  )

### CODE REVIEW

File: offline/packages/CaloReco/RawClusterPositionCorrection.cc
Comment (pinkenburg) [inline offline/packages/CaloReco/RawClusterPositionCorrection.cc:1]: This check is dangerous:
if (calibdir[0] == '/')
If there is no entry the cdb will return an empty string. But if we in the future decide to use local files a check for "/" will fail even for a valid calibration

Comment (ahodges21) [reply offline/packages/CaloReco/RawClusterPositionCorrection.cc:1]: ROOT has the TFile implementation IsZombie(), you think that's a better implementation?

**Comment by osbornjd** [comment]
The scan build update errors are from the Acts update, please ignore

**Comment by pinkenburg** [comment]
Allways one more thing - there is a memory leak by not deleting the cdbhistos after use (either once the historgam has been copied from it or in the dtor)

**Comment by bseidlit** [comment]
This is looking great! But are we sure we don't want to interpolate between the bins or something? Also do we want to get the energy boundaries for where the residual correction is aplied from the histo, then all we need to do is upload a new histo to extend the range?

**Comment by ahodges21** [comment]
> This is looking great! But are we sure we don't want to interpolate between the bins or something? 

I'd say absolutely, but I wanted to get what we used for QM up and running first before upgrading the performance. Right now the deviations between bins is super small, so I don't think it's a priority to change. 

Also do we want to get the energy boundaries for where the residual correction is aplied from the histo, then all we need to do is upload a new histo to extend the range?

You mean instead of doing "if(cluster energy > blah && cluster energy < blah)" just go by something like Hist -> GetBinEdge(lastBin)? We can certainly implement that.

**Pull request merged**